### PR TITLE
CI: make systemd tests parallel-safe (*)

### PR DESCRIPTION
**Pull request by edsantiago**
Mostly just switch to safename. Rewrite setup() to guarantee
unique service file names, atomically created.

* IMPORTANT NOTE: enabling parallelization on these tests
  triggers #24010 ("fragment file" flake), but only on my
  f40 laptop. I have never seen the flake in Cirrus despite
  many many runs in #23275. I am submitting this for review
  and merging because even though _something_ is broken,
  this breakage is unlikely to affect our CI.

Signed-off-by: Ed Santiago <santiago@redhat.com>
```release-note
None
```

### CODE REVIEW

File: test/system/250-systemd.bats [hunk 1]
```diff
@@ -40,12 +46,13 @@ function service_setup() {
...
     (
-        cd $UNIT_DIR
+        cd $PODMAN_TMPDIR
         run_podman generate systemd --files --name \
                -e http_proxy -e https_proxy -e no_proxy \
                -e HTTP_PROXY -e HTTPS_PROXY -e NO_PROXY \
                --new $cname
-        mv "container-$cname.service" $UNIT_FILE
+        mv -Z "container-$cname.service" $UNIT_FILE.tmp.$$ && \
+            mv -Z $UNIT_FILE.tmp.$$ $UNIT_FILE
```
Comment (Luap99) [inline test/system/250-systemd.bats:55]: I am confused, why do we move to a tmp file just to move to the real location directly after? I do not see th epoint of doing this vs one single mv command?

File: test/system/250-systemd.bats [hunk 2]
```diff
@@ -230,14 +237,15 @@ LISTEN_FDNAMES=listen_fdnames" | sort)
...
     run_podman create --name $cname $IMAGE top
 
     # See note in service_setup() above re: using --files
     (
-        cd $UNIT_DIR
+        cd $PODMAN_TMPDIR
         run_podman generate systemd --template --files -n $cname
-        mv "container-$cname.service" $TEMPLATE_FILE
+        mv -Z "container-$cname.service" $TEMPLATE_FILE.tmp.$$ && \
+            mv -Z $TEMPLATE_FILE.tmp.$$ $TEMPLATE_FILE
```
Comment (Luap99) [inline test/system/250-systemd.bats:248]: same here

File: test/system/250-systemd.bats (hunk 1 above)
Comment (edsantiago) [reply test/system/250-systemd.bats:55]: Ultraparanoia. `mv` is not atomic across filesystems, so this is theoretically possible:
1. mv starts writing foo.service
2. another parallel job runs systemctl daemon-reload
3. systemd sees (incomplete) foo.service and does whatever systemd does with it and bad things happen
4. mv completes

Even with all this paranoia, #24010 still triggers (on my laptop), so I'm pretty sure that the fragment-file flake has nothing to do with corrupt systemd service files ... but I just want to eliminate every last possibility of test errors.

Comment (Luap99) [reply test/system/250-systemd.bats:55]: Ok got it but then please comment the reason here because it is not obvious otherwise.

**Review by Luap99 — APPROVED** [review]
/lgtm

**Pull request merged**